bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyScriptEngineConfiguration.java

```diff
@@ -294,13 +276,134 @@ synchronized void configureGems(ScriptEngine engine, boolean force) {
+        } else if (!gemfilePath.isAbsolute()) {
+            gemfilePath = Path.of(HOME_PATH_ABS).resolve(gemfilePath);
+        }
+
+        if (gemfilePath.toFile().exists()) {
+            return gemfilePath.toString();
+        }
+
+        String configuredGemfileString = String.join("\n", configuration.bundle_gemfile_content);
+        if (configuredGemfileString.isBlank()) {
+            return null;
+        }
+
+        gemfilePath = Path.of(UI_GEMFILE_PATH);
+        Path gemfileDirectory = gemfilePath.getParent();
+        if (!gemfileDirectory.toFile().exists()) {
+            logger.debug("Creating Gemfile directory {}", gemfileDirectory);
+            gemfileDirectory.toFile().mkdirs();
+        }
+
+        File gemfile = gemfilePath.toFile();
+        try {
+            List<String> headers = List.of( //
+                    "# This Gemfile is auto-generated by the JRuby Scripting add-on", //
+                    "# Do not edit this file directly.", //
+                    "# Edit the JRuby Scripting configuration 'bundle_gemfile_content' instead.", //
+                    "");
+
+            List<String> configuredGemfileContent = new ArrayList<>(headers);
+            configuredGemfileContent.addAll(configuration.bundle_gemfile_content);
+
+            if (gemfile.exists()) {
+                List<String> fileContent = Files.readAllLines(gemfilePath, StandardCharsets.UTF_8);
+                if (fileContent.equals(configuredGemfileContent)) {
+                    logger.debug("Gemfile already exists and is up to date.");
+                    return gemfilePath.toString();
+                }
+            }
+
+            Files.write(gemfilePath, configuredGemfileContent, StandardCharsets.UTF_8);
+            return gemfilePath.toString();
+        } catch (IOException e) {
+            logger.warn("Error creating/writing to Gemfile {}: {}", gemfilePath, e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * Require Bundler setup in the ScriptEngine
+     * 
+     * @param engine
+     */
+    public void bundlerSetup(ScriptEngine engine) {
+        String gemfilePath = getGemfilePath();
+        if (gemfilePath == null) {
+            logger.debug("No Gemfile is found or configured. Skipping Bundler setup.");
+            return;
+        }
+
+        String code = """
+                require 'jruby'
+                JRuby.runtime.instance_config.update_native_env_enabled = false
+                ENV['BUNDLE_GEMFILE'] = '%s'
+                require 'bundler/setup'
+                """.formatted(gemfilePath);
+
+        try {
+            logger.debug("Running Bundler on Gemfile {}", gemfilePath);
+            logger.trace("Bundler code:\n{}", code);
+            engine.eval(code);
+        } catch (ScriptException e) {
+            logger.warn("Error running Bundler setup: {}", e.getMessage());
+        }
+    }
+
+    public void bundlerInstall(ScriptEngine engine) {
```

Comment:
this method is unnecessary with the auto_install change above. though it could still make sense to change it to the equivalent of `bundle update`, and control it with the same config setting as is used to check for updates at all